pkg/quantile/agent.go

```diff
@@ -50,14 +54,24 @@ func (a *Agent) Reset() {
 }
 
 // Insert v into the sketch.
-func (a *Agent) Insert(v float64) {
-	a.Sketch.Basic.Insert(v)
+func (a *Agent) Insert(v float64, sampleRate float64) {
+	k := agentConfig.key(v)
+	if sampleRate == 1 {
+		a.Sketch.Basic.Insert(v)
+		a.Buf = append(a.Buf, k)
 
-	a.Buf = append(a.Buf, agentConfig.key(v))
-	if len(a.Buf) < agentBufCap {
-		return
+		if len(a.Buf) < agentBufCap {
+			return
+		}
+	} else {
+		n := uint(math.Ceil(1 / sampleRate))
+		a.Sketch.Basic.InsertN(v, n)
```

Comment:
Is it an issue that `n` has the same value when `sampleRate=0.9` or `sampleRate=0.5`?

Reply:
The idea here is to avoid ever having an n < 1. I don't know what the expectations are of users who are using high sample rates, e.g. 50%+. I could make change this so that sampleRate is guaranteed to be between 0 and 1 (is that guaranteed elsewhere?)